File: app/src/main/java/com/developeruz/tasklist/adapters/viewpager/BasicViewPager2Adapter.java
package com.developeruz.tasklist.adapters.viewpager;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentActivity;
import androidx.viewpager2.adapter.FragmentStateAdapter;
import java.util.List;

public class BasicViewPager2Adapter extends FragmentStateAdapter {

    private final List<Fragment> mFragments;

    public BasicViewPager2Adapter(FragmentActivity fa, List<Fragment> fragments) {
        super(fa);
        this.mFragments = fragments;

    }

    @NonNull
    @Override
    public Fragment createFragment(int i) {
        return mFragments.get(i);
    }

    @Override
    public int getItemCount() {
        return mFragments.size();
    }
}

File: app/src/main/java/com/developeruz/tasklist/ui/adapters/viewpager/BasicViewPager2Adapter.kt
package com.developeruz.tasklist.ui.adapters.viewpager

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter

class BasicViewPager2Adapter(fa: FragmentActivity?, private val mFragments: List<Fragment>) :
    FragmentStateAdapter(
        fa!!
    ) {
    override fun createFragment(i: Int): Fragment {
        return mFragments[i]
    }

    override fun getItemCount(): Int {
        return mFragments.size
    }
}